src/main/java/org/opensearch/ad/transport/ADHCImputeTransportAction.java

```diff
@@ -156,4 +159,46 @@ protected ADHCImputeNodeResponse nodeOperation(ADHCImputeNodeRequest nodeRequest
         }
     }
 
+    /**
+     * Determines whether the model state should be processed based on various conditions.
+     *
+     * Conditions checked:
+     * - The model's last seen execution end time is not the minimum Instant value.
+     * - The current execution end time is greater than or equal to the model's last seen execution end time,
+     *   indicating that the model state was updated in previous intervals.
+     * - The entity associated with the model state is present.
+     * - The owning node for real-time processing of the entity, with the same local version, is present in the hash ring.
+     * - The owning node for real-time processing matches the current local node.
```

Comment:
before we could have occasions where two nodes have the same model for the same entity, so two nodes at the same time will find if there is an anomaly on the interval?

Reply:
yes.

Reply:
would this mean we would write result twice or we handled it later to only be written once?

Reply:
We would only write it once (not guaranteed) as here we only process imputation on a node the entity is mapped to.